From a7cfb1187a783803d535fab28c87468af3a30a59 Mon Sep 17 00:00:00 2001
From: Florian Bezdeka <florian.bezdeka@siemens.com>
Date: Fri, 5 Sep 2025 11:38:07 +0200
Subject: [PATCH 0213/1013] x86: irq_pipeline: Fix build failure for
 !IRQ_PIPELINE but X86_FRED
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The POSTED_MSI_NOTIFICATION_VECTOR is present since 6.10, so
FIRST_SYSTEM_APIC_VECTOR was off by one resulting in the following build
warning:

./arch/x86/include/asm/irq_vectors.h:104:41: error: array index in initializer exceeds array bounds
  104 | #define POSTED_MSI_NOTIFICATION_VECTOR  0xeb
      |                                         ^~~~
arch/x86/entry/entry_fred.c:98:37: note: in definition of macro ‘SYSVEC’
   98 | #define SYSVEC(_vector, _function) [_vector - FIRST_SYSTEM_VECTOR] = fred_sysvec_##_function
      |                                     ^~~~~~~
arch/x86/entry/entry_fred.c:121:16: note: in expansion of macro ‘POSTED_MSI_NOTIFICATION_VECTOR’
  121 |         SYSVEC(POSTED_MSI_NOTIFICATION_VECTOR,  posted_msi_notification),
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./arch/x86/include/asm/irq_vectors.h:104:41: note: (near initialization for ‘sysvec_table’)
  104 | #define POSTED_MSI_NOTIFICATION_VECTOR  0xeb
      |                                         ^~~~
arch/x86/entry/entry_fred.c:98:37: note: in definition of macro ‘SYSVEC’
   98 | #define SYSVEC(_vector, _function) [_vector - FIRST_SYSTEM_VECTOR] = fred_sysvec_##_function
      |                                     ^~~~~~~
arch/x86/entry/entry_fred.c:121:16: note: in expansion of macro ‘POSTED_MSI_NOTIFICATION_VECTOR’
  121 |         SYSVEC(POSTED_MSI_NOTIFICATION_VECTOR,  posted_msi_notification),
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Required build configuration:
  - CONFIG_IRQ_PIPELINE = n
  - CONFIG_X86_FRED = y

Signed-off-by: Florian Bezdeka <florian.bezdeka@siemens.com>
---
 arch/x86/include/asm/irq_vectors.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/include/asm/irq_vectors.h b/arch/x86/include/asm/irq_vectors.h
index de7592d870e9..dd755c95f3b4 100644
--- a/arch/x86/include/asm/irq_vectors.h
+++ b/arch/x86/include/asm/irq_vectors.h
@@ -109,7 +109,7 @@
 #define FIRST_SYSTEM_APIC_VECTOR	CALL_FUNCTION_OOB_VECTOR
 #define NR_APIC_VECTORS	        	(NR_VECTORS - FIRST_SYSTEM_VECTOR)
 #else
-#define FIRST_SYSTEM_APIC_VECTOR	LOCAL_TIMER_VECTOR
+#define FIRST_SYSTEM_APIC_VECTOR	POSTED_MSI_NOTIFICATION_VECTOR
 #endif
 
 #define NR_VECTORS			 256
-- 
2.30.2

